packages/angular_devkit/build_angular/src/builders/web-test-runner/index.ts

```diff
@@ -0,0 +1,194 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { BuilderContext, BuilderOutput, createBuilder } from '@angular-devkit/architect';
+import type * as WebTestRunner from '@web/test-runner';
+import { promises as fs } from 'node:fs';
+import path from 'node:path';
+import { findTestFiles } from '../../utils/test-files';
+import { buildApplicationInternal } from '../application';
+import { OutputHashing } from '../browser-esbuild/schema';
+import { logBuilderStatusWarnings } from './builder-status-warnings';
+import { WtrBuilderOptions, normalizeOptions } from './options';
+import { Schema } from './schema';
+
+export default createBuilder(
+  async (schema: Schema, ctx: BuilderContext): Promise<BuilderOutput> => {
+    ctx.logger.warn(
+      'NOTE: The Web Test Runner builder is currently EXPERIMENTAL and not ready for production use.',
+    );
+    logBuilderStatusWarnings(schema, ctx);
+
+    const options = normalizeOptions(schema);
+    const testDir = 'dist/test-out';
+
+    // Parallelize startup work.
+    const [testFiles, wtr] = await Promise.all([
+      // Glob for files to test.
+      findTestFiles(options.include, options.exclude, ctx.workspaceRoot).then((files) =>
+        Array.from(files).map((file) => path.relative(process.cwd(), file)),
+      ),
+      // Dynamic import `@web/test-runner` because as an optional peer dep, it may not be installed.
+      import('@web/test-runner').catch(() => undefined),
```

Comment:
Since this is installed by the users using `import('@web/test-runner')` will have an incorrect context as it will be loaded relative to this file instead of the users workspace. 

```js
 const workspaceRootRequire = createRequire(workspaceRoot + '/');
 workspaceRootRequire('@web/test-runner');
```

Reply:
Good call. I'm curious why the trailing slash? You clearly added that intentionally and it does appear to be load-bearing in my testing, though I couldn't find any documentation on it. Does it treat that file path part as file and not a directory, or something like that?

Reply:
Yeah correct, it is because the `createRequire` first parameter is a filepath.